flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -123,7 +125,9 @@ void stop() {
 
 		private static final String JSON_FILE_ENDING = ".json";
 
-		JobArchiveFetcherTask(List<HistoryServer.RefreshLocation> refreshDirs, File webDir, CountDownLatch numFinishedPolls) {
+		JobArchiveFetcherTask(long retainedApplicationsMillis, List<HistoryServer.RefreshLocation> refreshDirs, File webDir,
+							CountDownLatch numFinishedPolls) {
+			this.retainedApplicationsMillis = retainedApplicationsMillis;
```

Comment:
Maybe we could add a test logic whether the passed in parameter `retainedApplicationsMillis` is positive or not here

Reply:
The test logic has been added by increasing the isCleanupEnabled variable. Thanks a lot for reviewing it again!